### ISSUE TITLE
Does datapasta work with RStudio Server?

### ISSUE THREAD

**Issue by cmhh** [MilesMcBain/datapasta#73]
Hi
This is cool.  However, at work we use RStudio Server.  When I try pasting with RStudio Server I just get the following:

`Clipboard is not available. Is xsel or xclip installed? Is DISPLAY set?`

It this a technical limitation with remote execution?  Will it only work on a local desktop?

Cheers

**Comment by MilesMcBain** [comment]
Yeah this problem arises since RStudio server is trying to access the
clipboard of the remote machine, not your local terminal where the data
resides.

I have heard workarounds are possible through Dan Wilson,
@thedatacollective, I'll see if he has anything to add here.

**Comment by kok00n** [comment]
Hi,

Do u have any update on this issue? I've got the same problem with RStudio Server.

BR

**Comment by psalveson** [comment]
Same issue here.

-P

**Comment by MilesMcBain** [comment]
Sorry friends. The approach used by Dan will not work with `datapasta`.

The RStudio API does not currently allow me to talk to your browser, which is what we would need to do to get your local clipboard. Datapasta currently looks on the server's clipboard, and often fails when the required Linux applications are not installed, as in the case of the OP.

**Comment by ablack3** [comment]
This would be a nice feature to add.

**Comment by jon-harmon-macmillan** [comment]
> `datapasta` currently looks on the server's clipboard, and **often fails when the required Linux applications are not installed,** as in the case of the OP.

Does it work if the correct applications ARE installed? I wasn't sure if the error message meant "xsel or xclip might be getting in the way" or "I need one of these for this to work." I'd love to get datapasta working with Rstudio server, and will do some heavy lifting if it will potentially work.

**Comment by MilesMcBain** [comment]
No, even if you have xsel or xclip you'll still be trying to read from the server's clipboard, not your local one via your browser. So it will appear to 'work' but won't be useful.

**Comment by jon-harmon-macmillan** [comment]
Good to know for an addin I had in mind for an internal package at work! I'll keep my eye on this in case a solution presents itself.
